### PR TITLE
feat: Send plugin logs via OTEL

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

~~Still WIP, opening for reference~~

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests 🧪
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

File: serve/opentelemetry.go [hunk 1]
```diff
@@ -146,20 +189,116 @@ func setupOtel(ctx context.Context, logger zerolog.Logger, p *plugin.Plugin, ote
...
+	ctx context.Context
+}
+
+func (h *otelLoggerHook) Run(e *zerolog.Event, level zerolog.Level, message string) {
+	record := otellog.Record{}
+	record.SetTimestamp(time.Now().UTC())
+	record.SetSeverity(otellogSeverity(level))
+	record.SetBody(otellog.StringValue(message))
+	// See https://github.com/rs/zerolog/issues/493, this is ugly but it works
+	logData := make(map[string]any)
```
Comment (erezrokah) [inline serve/opentelemetry.go:233]: Opened https://github.com/rs/zerolog/pull/682 let's see how that goes 😄

Comment (marianogappa) [reply serve/opentelemetry.go:233]: 😱 https://github.com/rs/zerolog/issues/493#issuecomment-1458241050 🤣

Comment (marianogappa) [reply serve/opentelemetry.go:233]: Can you just add a general comment (not about the hack, but about what's going on with zerolog + otel? It's very clear with the context of this PR but the code looks complex so it's not clear on its own I think.

**Review by marianogappa — APPROVED** [review]

**Comment by erezrokah** [comment]
Thanks for the review, added a more detailed comment. Someone is reviewing my PR at https://github.com/rs/zerolog/pull/682 but I wasn't able to understand if they are a maintainer or not.

**Pull request merged**